src/apify/storage_clients/_file_system/_key_value_store_client.py

```diff
@@ -0,0 +1,36 @@
+import asyncio
+
+from typing_extensions import override
+
+from crawlee._consts import METADATA_FILENAME
+from crawlee.storage_clients._file_system import FileSystemKeyValueStoreClient
+
+from apify._configuration import Configuration
+
+
+class ApifyFileSystemKeyValueStoreClient(FileSystemKeyValueStoreClient):
+    """Apify-specific implementation of the `FileSystemKeyValueStoreClient`.
+
+    The only difference is that it overrides the `purge` method to delete all files in the key-value store
+    directory, except for the metadata file and the `INPUT.json` file.
+    """
+
+    @override
+    async def purge(self) -> None:
```

Comment:
Does this override mean that Crawlee does not treat `INPUT.json` differently from other files anymore and this logic is only added in the SDK? I'm a fan, but I wonder if the rest of the team is on board with this.

Reply:
this is the part where we handle input file in the project root in JS

https://github.com/apify/crawlee/blob/5c4726df96e358a9bbf44a0cd2760e4e269f0fae/packages/core/src/storages/key_value_store.ts#L697

Reply:
From the "Crawlee is a web scraping framework" point of view, handling input feels a bit redundant here - if you intend to only run the crawler outside of Apify, you'll probably want to accept input some other way - CLI arguments, environment variables, some domain-speciffic config file... Providing a helper for loading a JSON file from two possible locations doesn't add much value IMO.

Asking Apifiers about this doesn't make much sense in this situation, because basically everything we write is intended to run on Apify :slightly_smiling_face: